File: Ruby/playground.rb
# @param {Integer} x
# @return {Boolean}

# simple palidrome code
def is_palindrome(x)
    if x.to_s[0] == "-"
        return false
    end
    if x == x.to_s.reverse.to_i
       return true
    end
    return false
end

# playing with classes and class methods
class Automobile
    def initialize(make, model, year)
        @make = make
        @model = model
        @year = year
    end
    def to_s
        "Automobile: #{@make}--#{@model} (#{@year})"
    end
    def Automobile.revIt 
        puts "vroom"
    end
end

# inheritance and instance methods
class Truck < Automobile
    def initialize(make, model, year, bedLength)
      super(make, model, year)
      @bedLength = bedLength
    end
    def revIt
        puts "vroom vroom"
    end
end

aTruck = Truck.new("2022", "Tesla", "WorkTruck", "10")
aTruck.revIt
puts "Done!"

File: Kotlin/playground.kt
// classes, inheritance, interfaces, class methods, abstract functions, member functions

abstract class Automobile(val make: String) {
    abstract fun revIt()
}

interface Hauler {
    fun unload()
    fun haul(loadSize: Int) = println("I'm going to haul this $loadSize ton load")
}

class Truck(make: String, val model: String) : Automobile(make), Hauler {
    override fun revIt() = println("vroom vroom")
    fun load() = println("Load me up!")
    override fun unload() = println("Unload It!")
}

fun main() {
    val newVehicle = Truck("Tesla", "Super Truck")
    newVehicle.revIt() // An implementation of an abstract function
    newVehicle.load() // A member function
    newVehicle.haul(10) // A default implementation in an interface
    for (i in 1..10) {           
        println(" $i")
    }
    newVehicle.unload() // An implementation of an interface function
}


